src/intelligence/browser/browser.service.ts

```
@@ -567,83 +612,246 @@ ${text}
     }
   }
 
-  async aiSearchStream(query: string, emitUpdate: (data: any) => void) {
-    emitUpdate({ status: 'started', message: 'Starting search...', query });
+  async aiSearchStream(
+    query: string,
+    userId: string,
+    emitCallback?: Function,
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)